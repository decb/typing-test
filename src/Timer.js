import React from "react";

import { gameStates } from "./gameStates";
import { useInterval } from "./useInterval";

const totalSeconds = 4;

function Timer(props) {
  const [seconds, setSeconds] = React.useState(() => totalSeconds);

  useInterval(() => {
    if (props.gameState === gameStates.DURING) {
      if (seconds === 0) {
        props.endGame();
      } else {
        setSeconds(seconds - 1);
      }
    }
  }, 1000);

  return <span>{seconds} seconds</span>;
}

export default Timer;
