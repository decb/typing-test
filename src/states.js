export const appStates = {
  NONE_SELECTED: 0,
  LOADING: 1,
  READY: 2,
  ERROR: 3
};

export const gameStates = {
  BEFORE: 0,
  DURING: 1,
  OVER: 2
};
