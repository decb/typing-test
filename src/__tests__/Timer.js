import React from "react";
import renderer from "react-test-renderer";

import Timer from "../Timer";
import { gameStates } from "../states";

describe("<Timer />", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(<Timer gameState={gameStates.BEFORE} setGameState={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
