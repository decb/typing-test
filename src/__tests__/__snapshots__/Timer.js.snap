// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> Renders correctly 1`] = `
<span>
  60
   seconds
</span>
`;
